lib/rspec/core/example_group.rb

```diff
@@ -557,11 +556,17 @@ def self.run_examples(reporter)
           instance = new(example.inspect_output)
           set_ivars(instance, before_context_ivars)
           succeeded = example.run(instance, reporter)
-          RSpec.world.wants_to_quit = true if fail_fast? && !succeeded
+          if !succeeded && fail_fast?
+            RSpec.world.wants_to_quit = true if reporter.fail_fast_limit_met?
+          end
           succeeded
         end.all?
       end
 
+      def self.fail_fast?
+        RSpec.configuration.fail_fast
+      end
+
```

Comment:
Any reason you moved this?  As far as I can see this isn't needed...and generally it's nice not to move code unless that's the point of the PR or commit.  You mind reverting it and rebasing it out of the commits in your PR?